client/network/src/request_responses.rs

```diff
@@ -982,9 +977,151 @@ mod tests {
 			}
 
 			match response_receiver.unwrap().await.unwrap().unwrap_err() {
-				super::RequestFailure::Network(super::OutboundFailure::ConnectionClosed) => {},
+				RequestFailure::Network(OutboundFailure::ConnectionClosed) => {},
 				_ => panic!()
 			}
 		});
 	}
+
+	/// [`RequestId`] is a monotonically increasing integer, starting at `1`. A [`RequestId`] is
```

Comment:
Strictly speaking, `RequestId` is an opaque type. I don't think the `libp2p-request-response` API or documentation states or exposes that it is necessarily a monotonically increasing integer. The only obvious property that it is expected to have besides those implied by the traits such as `Eq` and `Hash` that it implements, is that it uniquely identifies an inbound or outbound request in the context of the `RequestResponse` behaviour that returned it. Note that even for the same behaviour, there is no guarantee that inbound and outbound request IDs are distinct. In fact they use separate counters. It maybe should be stated more clearly in the documentation of `libp2p-request-response` what guarantees are _not_ given, especially that request IDs are not "globally" unique.

Reply:
> Strictly speaking, RequestId is an opaque type.

Good point. Agreed that the test doc comment should treat it that way. I updated the comment in https://github.com/paritytech/substrate/pull/7957/commits/c5717e09b5cb12c9a15589c41d49dca49a3d1b85. Let me know what you think.

> It maybe should be stated more clearly in the documentation of libp2p-request-response what guarantees are not given, especially that request IDs are not "globally" unique.

I will follow up with a pull request on rust-libp2p documenting this.